File: src/my_project_typescript/medium_problems_typescript/from51to100/findDuplicateNumber.ts
function findDuplicate(nums: number[]): number {

    let solutionObj: {[key:number]:number} = {}

    for (let item of nums){
        solutionObj[item] = 0
    }

    

    for (let item of nums){
        solutionObj[item] += 1
        if (solutionObj[item] >= 2){
            return item
        }

    }

};

File: src/my_project/easy_problems/from251to300/find_duplicate_number.py
from typing import List 

class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        
        solution_dict = dict()
        
        for item in nums:
            
            if item not in solution_dict:
                solution_dict[item] = 0 

        for item in nums: 
            solution_dict[item] += 1
            if solution_dict[item] >=2:
                return item 
            
            
        
            
            
        